flexget/plugins/filter/movie_queue.py

```diff
@@ -396,3 +432,302 @@ def queue_get(session=None, downloaded=False):
+
+movie_object = {
+    'type': 'object',
+    'properties': {
+        'added': {'type': 'string'},
+        'downloaded': {'type': 'string'},
+        'entry_original_url': {'type': 'string'},
+        'entry_title': {'type': 'string'},
+        'entry_url': {'type': 'string'},
+        'id': {'type': 'integer'},
+        'imdb_id': {'type': 'string'},
+        'quality': {'type': 'string'},
+        'title': {'type': 'string'},
+        'tmdb_id': {'type': 'string'},
+    }
+}
+
+movie_queue_schema = {
+    'type': 'object',
+    'properties': {
+        'movies': {
+            'type': 'array',
+            'items': movie_object
+        },
+        'number_of_movies': {'type': 'integer'},
+        'total_number_of_pages': {'type': 'integer'},
+        'page_number': {'type': 'integer'}
+
+    }
+}
+
+
+def movie_queue_sort_value_enum(value):
+    enum = ['added', 'downloaded', 'id', 'title']
+    if not isinstance(value, basestring):
+        raise ValueError('Value expected to be string')
+    if value not in enum:
+        raise ValueError('Value expected to be in' + ' ,'.join(enum))
+    return value
+
+
+def movie_queue_sort_order_enum(value):
+    enum = ['desc', 'asc']
+    if not isinstance(value, basestring):
+        raise ValueError('Value expected to be string')
+    if value not in enum:
+        raise ValueError('Value expected to be in' + ' ,'.join(enum))
+    if value == 'desc':
+        return True
+    return False
+
+
+movie_queue_schema = api.schema('list_movie_queue', movie_queue_schema)
+
+movie_queue_parser = api.parser()
+movie_queue_parser.add_argument('page', type=int, default=1, help='Page number')
+movie_queue_parser.add_argument('max', type=int, default=100, help='Movies per page')
+movie_queue_parser.add_argument('downloaded_only', type=inputs.boolean, default='false', help='Show only downloaded')
+movie_queue_parser.add_argument('sort_by', type=movie_queue_sort_value_enum, default='added',
+                                help="Sort response by 'added', 'downloaded', 'id', 'title'")
+movie_queue_parser.add_argument('order', type=movie_queue_sort_order_enum, default='desc', help='Sorting order')
+
+movie_add_results_schema = {
+    'type': 'object',
+    'properties': {
+        'message': {'type': 'string'},
+        'movie': movie_object
+    }
+}
+
+movie_add_input_schema = {
+    'type': 'object',
+    'properties': {
+        'title': {'type': 'string'},
+        'imdb_id': {'type': 'string', 'pattern': r'tt\d{7}'},
+        'tmdb_id': {'type': 'integer'},
+        'quality': {'type': 'string', 'format': 'quality_requirements', 'default': 'any'}
+    },
+    'anyOf': [
+        {'required': ['title']},
+        {'required': ['imdb_id']},
+        {'required': ['tmdb_id']}
+    ]
+}
+
+movie_add_results_schema = api.schema('movie_add_results', movie_add_results_schema)
+movie_add_input_schema = api.schema('movie_add_input_schema', movie_add_input_schema)
+
+movie_del_results_schema = {
```

Comment:
I don't think we need a schema for deletes? Just return a 200 with a message is fine

or.. we should come up with a standard results schema